src/sbd-inquisitor.c

```diff
@@ -1307,37 +1303,51 @@ int main(int argc, char **argv, char **envp)
         } else if (strcmp(argv[optind], "ping") == 0) {
             exit_status = ping_via_slots(argv[optind + 1], servants_leader);
 
-        } else if (strcmp(argv[optind], "watch") == 0) {
+        } else
+#endif
+        if (strcmp(argv[optind], "query-watchdog") == 0) {
+            exit_status = watchdog_info();
+        } else if (strcmp(argv[optind], "test-watchdog") == 0) {
+            exit_status = watchdog_test();
+        } else {
+            exit_status = -2;
+        }
+
+        if (strcmp(argv[optind], "watch") == 0) {
+            /* sleep $(sbd $SBD_DEVICE_ARGS dump | grep -m 1 msgwait | awk '{print $4}') 2>/dev/null */
+
+                const char *delay_source = "SBD_DELAY_START";
+
+#if SUPPORT_SHARED_DISK
                 if(disk_count > 0) {
                     /* If no devices are specified, its not an error to be unable to find one */
                     open_any_device(servants_leader);
                 }
 
+                if (delay_start && delay <= 0) {
+                    delay = get_first_msgwait(servants_leader);
+                    delay_source = "msgwait";
```

Comment:
probably just 'if (delay > 0)' as otherwise 2 * watchdog-timeout is taken (disk is probably not the source as otherwise reading msgwait would have worked).
even better would probably have been to bail out if msgwait can't be read but let's leave that to a maybe subsequent PR as it would mean a change in existing behavior and for 3 disks it shouldn't fail if the first disk fails.